tests/test_project.py

```diff
@@ -566,6 +633,16 @@ def test_get_contrast_model_field(input_calc: 'RAT.project.CalcTypes', input_mod
     assert project.get_contrast_model_field() == expected_field_name
 
 
+def test_write_script(test_project, test_project_script) -> None:
+    """Test the script we write to regenerate the project is as expected."""
+    mock_file = mock.mock_open()
+    with mock.patch('builtins.open', mock_file):
+        test_project.write_script()
+
+    # We check the calls to the write function as we do not return what we write
+    assert mock_file().write.call_args_list == test_project_script
+
+
```

Comment:
After doing this, is there a way we can run the script we have and compare the resulting object against "test_project". I've tried to put the script together as a string and use exec, but I can't return the object that way.